AVsitter2/Utilities/Missing-anim-finder.lsl

```diff
@@ -64,27 +66,38 @@ default
         {
             if (data == EOF)
             { 
-                if(NOT_FOUND_COUNT>0){
+                if(!NOT_FOUND_COUNT){
+                    Owner_Say("All animations referenced in the notecard were accounted for.");   
+                }
+                else{
                     Owner_Say("Anims were used in the notecard but not found in inventory!");
                 }
                 integer i;
                 for (i = 0; i < llGetInventoryNumber(INVENTORY_ANIMATION); i++)
                 {
+                    integer isVariableSpeed;
                     string anim_basename = llGetInventoryName(INVENTORY_ANIMATION, i);
                     if(llListFindList(["+","-"], [llGetSubString(anim_basename,-1,-1)]) != -1)
                     {
+                        isVariableSpeed=TRUE;
                         anim_basename = llGetSubString(anim_basename,0,-2);
```

Comment:
This has a wraparound hazard. An animation simply named "+" would result in anim_basename being "+" as well. The fix is replacing `llGetSubString` with `llDeleteSubString`.

Reply:
Do you propose to replace the line with:
```
if(llListFindList(["+","-"], [llDeleteSubString(anim_basename,0,-2)]) != -1)
```
If so, then I agree it looks like that would cut out the edge case of an anim named "-" or "+".

Still being a little unsteady with GitHub: Should I edit my original commit and re-submit?

Reply:
I'm proposing changing `anim_basename = llGetSubString(anim_basename,0,-2);` to `anim_basename = llDeleteSubString(anim_basename,-1,-1);` (I didn't have a chance to review this one before it went in).

This means that the animation name will be taken as completely empty. It's probably not a case that needs to be dealt with, but good practices dictate to avoid llGetSubString when it can possibly trim to the null string, as LSL will return the whole string instead in that case. I'm afraid the script may wrongly believe that the name of the animation is "+" and the suffix is "+" too, which is of course wrong.

Now that it's reviewed, please submit the changes as new commits, so that reviewers (me in this case) don't need to check the whole PR again from the beginning and can focus just on the corrections.

Reply:
I agree we should fix the issue you raise, but I'm still unsure about your suggestion here. 

Making the change as suggested will result an empty anim_basename, and give the message:

```
Animation '+' found in inventory but not used in notecard!
```

I think we'd want to avoid tagging an anim named "+" or "-" as a variable speed anim in the first place. That's why I assumed you were talking about changing the llGetSubString to llDeleteSubString in the if statement i.e:

```
if(llListFindList(["+","-"], [llGetSubString(anim_basename,-1,-1)]) != -1)
```

to

```
if(llListFindList(["+","-"], [llDeleteSubString(anim_basename,0,-2)]) != -1)
```

This will mean that the animation "+" won't be considered to be a variable-speed anim, anim_basename will remain the original name, and will not ever get to the ```anim_basename =``` code in question.
 
"+" can not be considered a varibale speed anim itself, while it's variable speed counterparts would be "++" and "+-".

Reply:
Doesn't that apply to every animation? You can have an animation called +MyAnim+ and a faster version called +MyAnim++, right?

This approach of classifying animations depending on their suffix doesn't look too robust to me. But if it's going to be used, at least let's use it consistently, rather than piling up hacks to work around its limitations in some corner cases. The whole strategy can be rewritten in future to match what the core does.

Reply:
> Doesn't that apply to every animation? You can have an animation called +MyAnim+ and a faster version called +MyAnim++, right?

Good point there. I've been working on the simplistic assumption that any anim ending in "+" or "-" was intended as a variable speed, but anims ending in "+" or "-" can also be considered valid base animations.

I'll take this into account with my next PR.